integration-tests/tests/base_test.go

```diff
@@ -70,73 +71,72 @@ func Test(t *testing.T) {
 	runner.TestingT(t, output)
 
 	if _, err := os.Stat(config.DefaultFileName); err == nil {
-		cfg, err := config.ReadConfig(config.DefaultFileName)
-		if err != nil {
-			t.Fatalf("Error reading config: %v", err)
-		}
-
-		if err := tearDownSnapd(cfg.FromBranch); err != nil {
-			t.Fatalf("Error stopping daemon: %v", err)
-		}
+		tearDownSnapd(&check.C{})
 	}
 }
 
 func setUpSnapd(c *check.C, fromBranch bool, extraEnv string) {
 	cli.ExecCommand(c, "sudo", "systemctl", "stop",
 		"snapd.service", "snapd.socket")
 
-	if fromBranch {
-		binPath, err := filepath.Abs("integration-tests/bin/snapd")
-		c.Assert(err, check.IsNil)
+	cli.ExecCommand(c, "sudo", "mkdir", "-p", cfgDir)
 
-		_, err = cli.ExecCommandErr("sudo", "mount", "-o", "bind",
-			binPath, daemonBinaryPath)
+	if fromBranch {
+		err := writeCoverageConfig()
 		c.Assert(err, check.IsNil)
 	}
 
 	err := writeEnvConfig(extraEnv)
 	c.Assert(err, check.IsNil)
 
-	_, err = cli.ExecCommandErr("sudo", "systemctl", "daemon-reload")
-	c.Assert(err, check.IsNil)
+	cli.ExecCommand(c, "sudo", "systemctl", "daemon-reload")
 
-	_, err = cli.ExecCommandErr("sudo", "systemctl", "start", "snapd.service")
-	c.Assert(err, check.IsNil)
+	cli.ExecCommand(c, "sudo", "systemctl", "start", "snapd.service")
 }
 
-func tearDownSnapd(fromBranch bool) error {
-	if _, err := cli.ExecCommandErr("sudo", "systemctl", "stop",
-		"snapd.service"); err != nil {
-		return err
-	}
+func tearDownSnapd(c *check.C) {
+	cli.ExecCommand(c, "sudo", "systemctl", "stop",
+		"snapd.service")
+
+	cli.ExecCommand(c, "sudo", "rm", "-rf", cfgDir)
+
+	cli.ExecCommand(c, "sudo", "systemctl", "daemon-reload")
 
-	if _, err := cli.ExecCommandErr("sudo", "rm", "-rf", cfgDir); err != nil {
+	cli.ExecCommand(c, "sudo", "systemctl", "start", "snapd.service")
+}
+
+// this function writes a config file for snapd.service which clears and overrides the default
+// ExecStart setting adding the required flags for recording coverage info
+func writeCoverageConfig() error {
+	cfgFileName := "coverage.conf"
+	cfgFile := filepath.Join(cfgDir, cfgFileName)
+
+	binPath, err := filepath.Abs("integration-tests/bin/snapd")
```

Comment:
a note for future: this could conflict with the rename branch. Marking it in my TODO to make sure we merge it alright depending on which lands first..